lib/gemstash/storage.rb

```diff
@@ -27,23 +27,22 @@ def initialize(folder, version)
     def initialize(folder, root: true)
       @folder = folder
       check_storage_version if root
-      FileUtils.mkpath(@folder) unless Dir.exist?(@folder)
     end
 
     # Fetch the resource with the given +id+ within this storage.
     #
     # @param id [String] the id of the resource to fetch
     # @return [Gemstash::Resource] a new resource instance from the +id+
     def resource(id)
-      Resource.new(@folder, id)
+      Resource.new(self.class.storage_service, @folder, id)
     end
 
     # Fetch a nested entry from this instance in the storage engine.
     #
     # @param child [String] the name of the nested entry to load
     # @return [Gemstash::Storage] a new storage instance for the +child+
     def for(child)
-      Storage.new(File.join(@folder, child), root: false)
+      self.class.new(File.join(@folder, child), root: false)
```

Comment:
🤔  I don't understand this change, why we need it? is the same as calling `Storage.new` ?